src/main/java/lol/ui/BattlefieldView.java

```diff
@@ -65,7 +65,32 @@ ImageView championView(Champion champion) {
 
   ImageView nexusView(Nexus nexus) {
     int teamID = nexus.teamOfNexus();
-    return(sprites.nexusesView(teamID));
+    double hpPercentage = (nexus.currentHP())/(double)(nexus.initialHP());
+    switch(nexus.teamOfNexus()) {
```

Comment:
The code in the Nexus.BLUE and Nexus.RED branches are the same. Try to find a way to group them together.
A way is to have an array of nexus, e.g.,
```java
private Image[] nexusOnFire;
private Image[] nexusDestroyed;
```

Then, `nexusDestroyed[Nexus.BLUE]` to access the blue / red images.